build error with gcc 7.3

```
[ 96%] Building C object zephyr/subsys/bluetooth/host/CMakeFiles/subsys__bluetooth__host.dir/keys.c.obj
cd /home/nashif/Work/github/zephyr/sanity-out/native_posix/tests/bluetooth/shell/test_br/zephyr/subsys/bluetooth/host && ccache /usr/lib64/ccache/gcc -DKERNEL -D_POSIX_C_SOURCE=200809 -D__ZEPHYR__=1 -I/home/nashif/Work/github/zephyr/kernel/include -I/home/nashif/Work/github/zephyr/arch/posix/include -I/home/nashif/Work/github/zephyr/arch/posix/soc/inf_clock -I/home/nashif/Work/github/zephyr/arch/posix/soc/inf_clock/include -I/home/nashif/Work/github/zephyr/arch/posix/soc/include -I/home/nashif/Work/github/zephyr/boards/posix/native_posix -I/home/nashif/Work/github/zephyr/include -I/home/nashif/Work/github/zephyr/include/drivers -I/home/nashif/Work/github/zephyr/sanity-out/native_posix/tests/bluetooth/shell/test_br/zephyr/include/generated -I/home/nashif/Work/github/zephyr/ext/lib/crypto/tinycrypt/include -I/home/nashif/Work/github/zephyr/subsys/bluetooth -I/home/nashif/Work/github/zephyr/tests/include  -m32   -O0 -g -Wall -Wformat -Wformat-security -Wno-format-zero-length -imacros /home/nashif/Work/github/zephyr/sanity-out/native_posix/tests/bluetooth/shell/test_br/zephyr/include/generated/autoconf.h -ffreestanding -Wno-main -std=c99 -Werror -DCONFIG_ASSERT=1 -D__ASSERT_ON=2 -Wno-deprecated-declarations -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -Wno-pointer-sign -Wno-unused-but-set-variable -fno-reorder-functions -fno-defer-pop -Werror=implicit-int -ffunction-sections -fdata-sections -Wno-format-truncation -fno-freestanding -Wno-undef -Wno-implicit-function-declaration -m32 -MMD -MP -include /home/nashif/Work/github/zephyr/arch/posix/include/posix_cheats.h -fprofile-arcs -ftest-coverage -o CMakeFiles/subsys__bluetooth__host.dir/keys.c.obj   -c /home/nashif/Work/github/zephyr/subsys/bluetooth/host/keys.c
/home/nashif/Work/github/zephyr/subsys/bluetooth/host/conn.c: In function ‘bt_conn_create_br’:
/home/nashif/Work/github/zephyr/subsys/bluetooth/host/conn.c:257:11: error: statement will never be executed [-Werror=switch-unreachable]
    return conn;
           ^~~~
/home/nashif/Work/github/zephyr/subsys/bluetooth/host/conn.c: In function ‘bt_conn_create_sco’:
/home/nashif/Work/github/zephyr/subsys/bluetooth/host/conn.c:309:11: error: statement will never be executed [-Werror=switch-unreachable]
    return sco_conn;
           ^~~~~~~~
cc1: all warnings being treated as errors
make[3]: *** [zephyr/subsys/bluetooth/host/CMakeFiles/subsys__bluetooth__host.dir/build.make:255: zephyr/subsys/bluetooth/host/CMakeFiles/subsys__bluetooth__host.dir/conn.c.obj] Error 1
```

--- a/subsys/bluetooth/host/conn.c
+++ b/subsys/bluetooth/host/conn.c
@@ -254,7 +254,6 @@ struct bt_conn *bt_conn_create_br(const bt_addr_t *peer,
 	conn = bt_conn_lookup_addr_br(peer);
 	if (conn) {
 		switch (conn->state) {
-			return conn;
 		case BT_CONN_CONNECT:
 		case BT_CONN_CONNECTED:
 			return conn;
@@ -306,7 +305,6 @@ struct bt_conn *bt_conn_create_sco(const bt_addr_t *peer)
 	sco_conn = bt_conn_lookup_addr_sco(peer);
 	if (sco_conn) {
 		switch (sco_conn->state) {
-			return sco_conn;
 		case BT_CONN_CONNECT:
 		case BT_CONN_CONNECTED:
 			return sco_conn;

